### PR TITLE
E2201. Testing for assignment_questionnaire_controller

### PR DESCRIPTION
**Pull request by jmhardy3**
Please set the title of your pull request in the following format: `[Project id/Independent study/Issue fix/...]. project name`, eg., `E904. Integrating courses and assignments on Expertiza with Moodle`

We suggest you including the below information in your pull request:
- A description of the changes proposed in the pull request.
- @mentions of the person or team responsible for reviewing proposed changes.
---
About Expertiza Bot
- If you have any questions about comments given by the expertiza-bot [(example)](https://github.com/expertiza/expertiza/pull/1877#issuecomment-762412918), you could create a comment in your pull request with the format `/dispute [UUID1] [UUID2]` [(example)](https://github.com/expertiza/expertiza/pull/1877#issuecomment-762430057) then the professor and TAs will be notified.
- [Here is a video](https://tinyurl.com/internet-bots) about how to communicate to expertiza-bot.

### CODE REVIEW

**Review by KrishnaSatya10 — APPROVED** [review]

**Review by KrishnaSatya10 — APPROVED** [review]

**Review by KrishnaSatya10 — APPROVED** [review]

**Review by KrishnaSatya10 — APPROVED** [review]

**Comment by jmhardy3** [comment]
/dispute [UUID1] [UUID2] is there something wrong with our Travis config?

File: app/controllers/assignment_questionnaire_controller.rb [hunk 1]
```diff
@@ -21,44 +21,53 @@ def action_allowed?
...
+      flash[:error] = 'Missing questionnaire ID - Questionnaire ID entered is Nil'
       return
-    end
-
-    assignment = Assignment.find(params[:assignment_id])
-    if assignment.nil?
+    
+    elsif assignment.nil?
       flash[:error] = 'Assignment #' + assignment.id + ' does not currently exist.'
       return
```
Comment (nnhimes) [inline app/controllers/assignment_questionnaire_controller.rb:54]: Isn't this the problem you fixed on line 24

File: app/controllers/assignment_questionnaire_controller.rb [hunk 2]
```diff
@@ -21,44 +21,53 @@ def action_allowed?
...
-    end
 
-    questionnaire = Questionnaire.find(params[:questionnaire_id])
-    if questionnaire.nil?
+    elsif questionnaire.nil?
       flash[:error] = 'Questionnaire #' + questionnaire.id + ' does not currently exist.'
```
Comment (nnhimes) [inline app/controllers/assignment_questionnaire_controller.rb:57]: Same here

File: app/controllers/assignment_questionnaire_controller.rb [hunk 3]
```diff
@@ -21,44 +21,53 @@ def action_allowed?
...
 
+    questionnaire = Questionnaire.find(params[:questionnaire_id])
+    assignment = Assignment.find(params[:assignment_id])
     @assignment_questionnaire = AssignmentQuestionnaire.new(params)
     @assignment_questionnaire.save
 
     respond_to do |format|
       format.json { render json: @assignment_questionnaire }
     end
   end
```
Comment (nnhimes) [inline app/controllers/assignment_questionnaire_controller.rb:70]: Else block should be indented and ends lined up right

**Comment by jmhardy3** [comment]
This is the error that is failing half of our tests:
image

File: Gemfile [hunk 4]
```diff
@@ -107,4 +107,4 @@ group :development, :test do
   gem 'capybara'
   gem 'selenium-webdriver'
   gem 'simplecov', require: false
-end
+end
```
Comment (nnhimes) [inline Gemfile:110]: Remember to fix before tonight

**Pull request merged**